### PR TITLE
[Lexer] Disable all SIL_KEYWORDs in non-SIL mode

### PR DESCRIPTION
**Pull request by rintaro**
The crash was caused by absence of `tok::kw_sil_scope` in the conditions.
Instead of long `||` conditions, use `SIL_KEYWORD` in Tokens.def.

Fixes a couple of crashers.

### CODE REVIEW

File: validation-test/compiler_crashers_fixed/28617-child-source-range-not-contained-within-its-parent-guard-stmt.swift [hunk 1]
```diff
@@ -5,6 +5,6 @@
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
-// RUN: not --crash %target-swift-frontend %s -emit-ir
+// RUN: not %target-swift-frontend %s -emit-ir
 {{guard let c:
 sil_scope
```
Comment (rintaro) [inline validation-test/compiler_crashers_fixed/28617-child-source-range-not-contained-within-its-parent-guard-stmt.swift]: This is accidental resolution.
There must be a range containment problem in the Parser.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**